### PR TITLE
chore(deps): remove recompose dependency

### PR DESCRIPTION
**Pull request by claudiu-muresan-pfa**
Reason behind the change is to partially fix remaining dependabot alerts:
- [node-fetch is vulnerable to Exposure of Sensitive Information to an Unauthorized Actor](https://github.com/waylayio/console/security/dependabot/44)
- [The `size` option isn't honored after following a redirect in node-fetch](https://github.com/waylayio/console/security/dependabot/7)

### CODE REVIEW

**Comment by francisdb** [comment]
Removed myself as reviewer as my react knowledge is very limited, of course always 👍 for removing dead dependencies

**Comment by SooiDP** [comment]
I don't see the point of removing recompose if you're not going to rewrite it with hooks, as the author suggests instead?

**Comment by claudiu-muresan-pfa** [comment]
Correct. It's just a "temporary stage" until we rewrite it with hooks. The main purpose was to remove recompose to fix the dependabot alerts on console. If you're ok with it, I'll create an issue for the hooks migration.

File: src/components/button/Group.styles.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ const base = (props: IButtonGroupProps) => css`
     border-right-width: 0;
   }
 
-  button:first-child, a:first-child {
+  button:first-of-type, a:first-of-type {
```
Comment (SooiDP) [inline src/components/button/Group.styles.tsx:25]: why has this been changed?

File: src/components/form/Field.styles.jsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ const inline = css`
     flex-shrink: 0;
   }
 
-  label:first-child {
+  label:first-of-type {
```
Comment (SooiDP) [inline src/components/form/Field.styles.jsx:16]: why was this changed?

File: src/components/input/Group.jsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ const Group = styled.div`
     }
   }
 
-  > *:first-child {
+  > *:first-of-type {
```
Comment (SooiDP) [inline src/components/input/Group.jsx:57]: same

File: src/components/button/Group.styles.tsx (hunk 1 above)
Comment (claudiu-muresan-pfa) [reply src/components/button/Group.styles.tsx:25]: There was an error when running the tests:
```
 console.error
      The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type".

      at node_modules/@emotion/core/node_modules/@emotion/cache/dist/cache.cjs.dev.js:272:27
          at Array.forEach (<anonymous>)
      at Function.<anonymous> (node_modules/@emotion/core/node_modules/@emotion/cache/dist/cache.cjs.dev.js:267:35)
      at H (node_modules/@emotion/stylis/dist/stylis.cjs.dev.js:530:24)
      at B (node_modules/@emotion/stylis/dist/stylis.cjs.dev.js:574:15)

```

**Comment by francisdb** [comment]
@bo-sec do we have any best practices regarding dismissing of security issues? Do we also have to dismiss them elsewhere?
